src/scheduler/MpiContext.cpp

```diff
@@ -25,6 +25,9 @@ void MpiContext::createWorld(const faabric::Message& msg)
     worldId = (int)faabric::util::generateGid();
     logger->debug("Initialising world {}", worldId);
 
+    // Update the original message to contain the world ID
+    msg.set_mpiworldid(worldId);
```

Comment:
This is needed so all `getExecutingWorld` from the master rank in `faasm`, call `MpiWorldRegistry::getOrInitialiseWorld` with the updated `worldId`.